master/internal/root.go

```diff
@@ -17,17 +25,17 @@ func root(ctx *actor.Context) error {
 	case actor.ChildFailed:
 		switch msg.Child.Address() {
 		case sproto.K8sRMAddr, sproto.AgentRMAddr:
-			ctx.Log().WithField("crash", msg).Errorf(clusterCrashMessage)
+			os.Exit(exit(ctx))
 		case sproto.PodsAddr, sproto.AgentsAddr:
-			ctx.Log().WithField("crash", msg).Errorf(clusterCrashMessage)
+			os.Exit(exit(ctx))
 		}
 		return nil
 	case actor.ChildStopped:
 		switch msg.Child.Address() {
 		case sproto.K8sRMAddr, sproto.AgentRMAddr:
-			ctx.Log().WithField("crash", msg).Errorf(clusterCrashMessage)
+			os.Exit((exit(ctx)))
 		case sproto.PodsAddr, sproto.AgentsAddr:
-			ctx.Log().WithField("crash", msg).Errorf(clusterCrashMessage)
+			os.Exit(exit(ctx))
```

Comment:
let's just return an error instead. if we do this, `sErr := m.system.Ref.AwaitTermination()` in `core.go` will trigger and crash the system. i think i mentioned this but that i wasn't sure if it was wired up, check and it is.